src/setupCoreModule.ts

```diff
@@ -22,26 +20,12 @@ export function setupCoreModule(): void {
         get hostVersion() {
             return channel.hostVersion;
         },
-        registerHook: (hookName: string, callback: HookCallback) => channel.registerHook(hookName, callback),
-        setProgrammingModel: (programmingModel: ProgrammingModel) => {
-            // Log when setting the programming model, except for the initial default one (partially because the grpc channels aren't fully setup at that time)
-            if (channel.app.programmingModel) {
-                channel.log({
-                    message: `Setting Node.js programming model to "${programmingModel.name}" version "${programmingModel.version}"`,
-                    level: LogLevel.Information,
-                    logCategory: LogCategory.System,
-                });
-            } else {
-                channel.defaultProgrammingModel = programmingModel;
-            }
-            channel.app.programmingModel = programmingModel;
-        },
+        registerHook,
+        setProgrammingModel,
         getProgrammingModel: () => {
             return channel.app.programmingModel;
         },
-        registerFunction: (metadata: FunctionMetadata, callback: FunctionCallback) => {
-            return registerFunction(metadata, callback);
-        },
+        registerFunction,
```

Comment:
Was there a reason this second layer of indirection added here? Maybe something about calling `this`?

Reply:
I had the indirection because I had to pass `channel` in, which was fixed when I made `channel` global: https://github.com/Azure/azure-functions-nodejs-worker/pull/686/files#diff-13b181030e6532174ea055646adc264a6263de35e2336dbeb25767e3ee59eabbL43

You can run into problems related to `this`, but it should not apply here.